### PR TITLE
Partially migrating monorepo linting to microservices using turborepo

### LINKED ISSUE #2736: Start migrating our monorepo linting to our microservices using turborepo
**What would you like to be added**:
We are just focusing on moving our linting

We just need to add the lint script to the package.json 

Goal pnpm lint and should run turborepo

**Why would you like this to be added**:

### PR DESCRIPTION
**Pull request by tcvan0707**
## Issue This PR Addresses
Fixes #2736 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This PR would be partially migrating our monorepo linting to our microservices using turborepo

## Steps to test the PR

- You should be in the root folder of Telescope
- run `pnpm turbo run lint`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by aserputov** [comment]
@tcvan0707 why this is a draft PR?

**Comment by sirinoks** [comment]
> @tcvan0707 why this is a draft PR?
This is work in progress

**Comment by tcvan0707** [comment]
> @tcvan0707 why this is a draft PR?

I need to get the reviews from reviewers first, I just added lint for `src/web`, not other parts yet. So I make this draft to get reviews

File: src/web/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      node: {},
+    },
+    react: {
+      version: '17.0',
+    },
+  },
+  overrides: [
+    // TypeScript for Next.js
+    {
+      files: ['src/web/**/*.ts', 'src/web/**/*.tsx'],
```
Comment (humphd) [inline src/web/.eslintrc.js:33]: I think you need to remove this bit from the root `.eslintrc.js` file, and only define the `src/web` stuff here.

**Review by humphd — CHANGES_REQUESTED** [review]
We need to update the way linting happens in CI to use turbo repo or something?

```
> @senecacdot/telescope@2.5.4 eslint /home/runner/work/telescope/telescope
> eslint --config .eslintrc.js "**/*.{jsx,tsx,ts,js}"


Oops! Something went wrong! :(

ESLint: 7.32.0

ESLint couldn't determine the plugin "jest" uniquely.

- /home/runner/work/telescope/telescope/node_modules/.pnpm/eslint-plugin-jest@25.3.4_cf0e947377590fbec5e3edece1a05d8c/node_modules/eslint-plugin-jest/lib/index.js (loaded in "--config")
- /home/runner/work/telescope/telescope/node_modules/.pnpm/eslint-plugin-jest@25.3.4_9a712f10e802afc11b6eaaed0fb6f7e4/node_modules/eslint-plugin-jest/lib/index.js (loaded in "src/web/.eslintrc.js")

Please remove the "plugins" setting from either config or remove either plugin installation.

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.

 ELIFECYCLE  Command failed with exit code 2.
Error: Process completed with exit code 1.
```

cc @menghif

File: src/web/.eslintrc.js (hunk 1 above)
Comment (tcvan0707) [reply src/web/.eslintrc.js:33]: Isn't it going through all the files with `*.ts` and `*.tsx` in `src/web`? I'm not sure what should be removed in here

Comment (humphd) [reply src/web/.eslintrc.js:33]: There is no `src/web` inside `src/web`, which is where you are right now.

The root `.eslintrc.js` needs to not also double-add this override (i.e., you are doing it here).

**Comment by menghif** [comment]
> We need to update the way linting happens in CI to use turbo repo or something?
> cc @menghif

Yes, I think we should use `pnpm turbo run lint` in CI.

I was helping @tcvan0707 with this PR and asked him to post it as a draft so we could all see it and contribute.

**Comment by menghif** [comment]
```
~/repos/telescope (issue-2736) % pnpm turbo run lint
• Packages in scope: @senecacdot/autodeployment, @senecacdot/feed-discovery-service, @senecacdot/image-service, @senecacdot/planet-service, @senecacdot/posts-service, @senecacdot/search-service, @senecacdot/sso-service, @senecacdot/status-service, @senecacdot/telescope-frontend, migrate, parser, users-microservice
• Running lint in 12 packages
@senecacdot/telescope-frontend:lint: cache miss, executing 83b7a9622a13c1cf
@senecacdot/telescope-frontend:lint: 
@senecacdot/telescope-frontend:lint: > @senecacdot/telescope-frontend@3.0.0 lint /Users/francesco/repos/telescope/src/web
@senecacdot/telescope-frontend:lint: > pnpm eslint
@senecacdot/telescope-frontend:lint: 

 Tasks:    1 successful, 1 total
Cached:    0 cached, 1 total
  Time:    1.401s 
```

When running this, Turborepo looks inside all the packages defined in `pnpm-workspace.yaml` and runs the `lint` command. The problem is that it does not run `lint` in the root of the project.

Can we get away from having to run `lint` in the root all together once [#2777](https://github.com/Seneca-CDOT/telescope/issues/2777) is done?

**Comment by humphd** [comment]
Yes, we can abandon the root directory and only do things in sub-projects.

**Comment by humphd** [comment]
@tcvan0707 how is this coming?

**Comment by menghif** [comment]
> There's another way we could approach this, if it's going to be hard to get this resolution to work: similar to how we work with Satellite now, we could publish the Telescope eslint config to npm, and have `pnpm` include it as a versioned module from the registry. This would also solve the Dockerfile problem.

@tcvan0707 and I are going to implement this in PR: https://github.com/Seneca-CDOT/telescope/pull/2862

Closing this one.

**Pull request closed without merging**